git_machete/generated_docs.py

```diff
@@ -0,0 +1,938 @@
+        If <branch> is provided but no local branch with the given name exists:
+        * if a remote branch of the same name exists in exactly one remote, then user is asked whether to check out this branch locally (as in `git checkout`),
+        * otherwise, user is asked whether it should be created as a new local branch.
+
+        If the definition file is empty or `-R/--as-root` is provided, the branch will be added as a root of the tree of branch dependencies.
+        Otherwise, the desired upstream (parent) branch can be specified with `-o/--onto`.
+        Neither of these options is mandatory, however; if both are skipped, git machete will try to automatically infer the target upstream.
+        If the upstream branch can be inferred, the user will be presented with inferred branch and asked to confirm.
+
+        Note: all the effects of `add` (except git branch creation) can as well be achieved by manually editing the definition file.
+
+        <b>Options:</b>
+           <b>-o</b>, <b>--onto=<target-upstream-branch></b>
+              Specifies the target parent branch to add the given branch onto. Cannot be specified together with `-R/--as-root`.
+           <b>-R</b>, <b>--as-root</b>
+              Add the given branch as a new root (and not onto any other branch). Cannot be specified together with `-o/--onto`.
+           <b>-y</b>, <b>--yes</b>
+              Don't ask for confirmation whether to create the branch or whether to add onto the inferred upstream.
+   """,
+    "advance": """
+        <b>Usage:</b><b>
+          git machete advance [-y|--yes]</b>
+
+        Fast forwards (as in `git merge --ff-only`) the current branch `C` to match its downstream `D`, pushes `C`
+        and subsequently slides out `D`. All three steps require manual confirmation unless `-y/--yes` is provided.
+
+        The downstream `D` is selected according to the following criteria:
+        * if `C` has exactly one downstream (child) branch `d` connected with a <green>green edge</green> (see status for definition) to `C` or is overridden, then `d` is selected as `D`,
+        * if `C` has no downstream branches connected with a <green>green edge</green> to `C`, then `advance` fails,
+        * if `C` has more than one downstream branch connected with a <green>green edge</green> to `C`,
+        then user is asked to pick the branch to fast-forward merge into (similarly to what happens in `git machete go down`). If `--yes` is specified, then `advance` fails.
+
+        As an example, if `git machete status --color=never --list-commits` is as follows:
+        <dim>
+          master
+          |
+          m-develop *
+            |
+            | Enable adding remote branch in the manner similar to git checkout
+            o-feature/add-from-remote
+            | |
+            | | Add support and sample for machete-post-slide-out hook
+            | o-feature/post-slide-out-hook
+            |
```

Comment:
Actually.. while we're here, I'd simplify this example. Let's just remove `chore/v3` and `feature/types` completely from both trees, they only obfuscate the point while adding little value